test/test_testing.py

```diff
@@ -1055,68 +1016,81 @@ def test_mapping_mismatching_values_msg(self, device):
+    def test_type_inequality(self):
+        actual = torch.empty(2)
         expected = actual.tolist()
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegexs(UsageError, str(type(actual)), str(type(expected))):
                 fn()
 
-    @onlyCPU
-    def test_unknown_type(self, device):
+    def test_unknown_type(self):
         actual = "0"
         expected = "0"
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegexs(UsageError, str(type(actual))):
                 fn()
 
-    @onlyCPU
-    def test_numpy(self, device):
-        tensor = torch.rand(2, 2, dtype=torch.float32, device=device)
+    def test_numpy(self):
+        tensor = torch.rand(2, 2, dtype=torch.float32)
         actual = tensor.numpy()
         expected = actual.copy()
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             fn()
 
-    @onlyCPU
-    def test_scalar(self, device):
-        tensor = torch.rand(1, device=device)
+    def test_scalar(self):
+        tensor = torch.rand(1)
         actual = expected = tensor.item()
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             fn()
 
-    @onlyCPU
-    def test_msg_str(self, device):
+    def test_msg_str(self):
         msg = "Custom error message!"
 
-        actual = torch.tensor(1, device=device)
-        expected = torch.tensor(2, device=device)
+        actual = torch.tensor(1)
+        expected = torch.tensor(2)
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegex(AssertionError, msg):
                 fn(msg=msg)
 
-    @onlyCPU
-    def test_msg_callable(self, device):
+    def test_msg_callable(self):
         msg = "Custom error message!"
 
         def make_msg(actual, expected, trace):
             return msg
 
-        actual = torch.tensor(1, device=device)
-        expected = torch.tensor(2, device=device)
+        actual = torch.tensor(1)
+        expected = torch.tensor(2)
 
         for fn in self.assert_fns_with_inputs(actual, expected):
             with self.assertRaisesRegex(AssertionError, msg):
                 fn(msg=make_msg)
 
 
-instantiate_device_type_tests(TestAsserts, globals())
+class TestAssertsMultiDevice(TestCase, _TestAssertsMixin):
+    @onlyCUDA
+    def test_mismatching_device(self, device):
+        actual = torch.empty((), device=device)
+        expected = actual.clone().cpu()
+
+        for fn in self.assert_fns_with_inputs(actual, expected):
+            with self.assertRaisesRegex(AssertionError, "device"):
+                fn()
+
+    @onlyCUDA
+    def test_mismatching_device_no_check(self, device):
+        actual = torch.rand((), device=device)
+        expected = actual.clone().cpu()
+
+        for fn in self.assert_fns_with_inputs(actual, expected):
+            fn(check_device=False)
+
+
+instantiate_device_type_tests(TestAssertsMultiDevice, globals())
```

Comment:
nit: you can also do 
```suggestion
instantiate_device_type_tests(TestAssertsMultiDevice, globals(), only_for='cuda')
```
so you dont need to explicitly set `@onlyCUDA` decorator for each tests above